src/Microsoft.ML.Data/Prediction/CalibratorCatalog.cs

```diff
@@ -0,0 +1,438 @@
+    typeof(SignatureLoadModel), "", NaiveCalibratorTransformer.LoadName)]
+
+[assembly: LoadableClass(typeof(CalibratorTransformer<PavCalibrator>), typeof(PavCalibratorTransformer), null,
+    typeof(SignatureLoadModel), "", PavCalibratorTransformer.LoadName)]
+
+namespace Microsoft.ML.Calibrator
+{
+
+    /// <summary>
+    /// Base class for CalibratorEstimators. CalibratorEstimators take a the result of a <see cref="BinaryClassifierScorer"/>, so an <see cref="IDataView"/>
+    /// that contains a "Score" column, and converts the scores to probabilities(through binning, interpolation etc.), based on the <typeparamref name="TICalibrator"/> type.
+    /// </summary>
+    /// <example>
+    /// <format type="text/markdown">
+    /// <![CDATA[
+    ///  [!code-csharp[Calibrators](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Calibrator.cs)]
+    /// ]]></format>
+    /// </example>
+    public abstract class CalibratorEstimatorBase<TCalibratorTrainer, TICalibrator> : IEstimator<CalibratorTransformer<TICalibrator>>
+        where TCalibratorTrainer : ICalibratorTrainer
+        where TICalibrator : class, ICalibrator
+    {
+        protected readonly IHostEnvironment Host;
+        protected readonly TCalibratorTrainer CalibratorTrainer;
+
+        protected readonly IPredictor Predictor;
+        protected readonly SchemaShape.Column ScoreColumn;
+        protected readonly SchemaShape.Column FeatureColumn;
+        protected readonly SchemaShape.Column LabelColumn;
+        protected readonly SchemaShape.Column WeightColumn;
+        protected readonly SchemaShape.Column PredictedLabel;
+
+        protected CalibratorEstimatorBase(IHostEnvironment env,
+            TCalibratorTrainer calibratorTrainer,
+            IPredictor predictor = null,
+            string labelColumn = DefaultColumnNames.Label,
+            string featureColumn = DefaultColumnNames.Features,
+            string weightColumn = null)
+        {
+            Host = env;
+            Predictor = predictor;
+            CalibratorTrainer = calibratorTrainer;
+
+            ScoreColumn = TrainerUtils.MakeR4ScalarColumn(DefaultColumnNames.Score); // Do we fantom this being named anything else (renaming column)? Complete metadata?
+            LabelColumn = TrainerUtils.MakeBoolScalarLabel(labelColumn);
+            FeatureColumn = TrainerUtils.MakeR4VecFeature(featureColumn);
+            PredictedLabel = new SchemaShape.Column(DefaultColumnNames.PredictedLabel,
+                SchemaShape.Column.VectorKind.Scalar,
+                BoolType.Instance,
+                false,
+                new SchemaShape(MetadataUtils.GetTrainerOutputMetadata()));
+
+            if (weightColumn != null)
+                WeightColumn = TrainerUtils.MakeR4ScalarWeightColumn(weightColumn);
+        }
+
+        SchemaShape IEstimator<CalibratorTransformer<TICalibrator>>.GetOutputSchema(SchemaShape inputSchema)
```

Comment:
Please add a doc string for this function --- For calibrator, the output  schema is computed by adding one probability column into input schema. Existing probability column may be overwritten.

Reply:
it won't be overitten. We will have two :)

---
In reply to: [240338011](https://github.com/dotnet/machinelearning/pull/1816#discussion_r240338011) [](ancestors = 240338011)